test/failjure/test_core.clj

```diff
@@ -63,31 +63,47 @@
             ""
             (str "O")
             (str "k")
-            (str "!")))) 
+            (str "!"))))
 
       (is (= (fail "Not OK!")
              (attempt->
                ""
                (str "Not OK!")
                (fail)
                (str "kay-O!")
-               (reverse)))))
+               (reverse))))
+
+      (is (= (fail "Not OK!")
+             (attempt->
+              (fail "Not OK!")
+              (str "Not OK!")
+              (fail)
```

Comment:
If your intention was to test the new functionality, I believe the old implementation would also fail here with the string "Not OK!"